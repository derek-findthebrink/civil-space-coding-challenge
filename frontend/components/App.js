/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { ApolloProvider } from "@apollo/react-hooks";
import { Helmet } from "react-helmet";
import { client } from "../graphql/client";

// Components
import Home from "../features/home/components/Home";
import ArticlesContainer from "../features/articles/components/ArticlesContainer";
import ArticleContainer from "../features/articles/components/ArticleContainer";
import { Layout } from "./Layout";

// Routes
const ROOT = "/";
const ARTICLES = "/articles";
const ARTICLE = "/article/:id";

export const Routes = () => (
  <ApolloProvider client={client}>
    <Switch>
      <Route
        strict
        path={ARTICLE}
        render={(props) => <ArticleContainer {...props} />}
      />
      <Route
        strict
        path={ARTICLES}
        render={(props) => <ArticlesContainer {...props} />}
      />
      <Route strict path={ROOT} render={(props) => <Home {...props} />} />
    </Switch>
  </ApolloProvider>
);

export const App = () => {
  return (
    <>
      <Helmet>
        <link rel="preconnect" href="https://fonts.gstatic.com" />
        <link
          href="https://fonts.googleapis.com/css2?family=Source+Serif+Pro&family=Open+Sans:wght@400;700&display=swap"
          rel="stylesheet"
        />
      </Helmet>
      <Layout>
        <BrowserRouter>
          <Routes />
        </BrowserRouter>
      </Layout>
    </>
  );
};

export default App;
