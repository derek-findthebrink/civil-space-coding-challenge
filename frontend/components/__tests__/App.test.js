import * as React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

import { Routes, App } from "../App";


jest.mock("../../features/home/components/Home", () => {
  return {
    __esModule: true,
    default: () => {
      return <h1>I am Home</h1>;
    },
  };
});

jest.mock('../../features/articles/components/ArticlesContainer', () => {
  return {
    __esModule: true,
    default: () => {
      return (<h1>I am ArticlesContainer</h1>)
    }
  }
})

jest.mock("../../features/articles/components/ArticleContainer", () => {
  return {
    __esModule: true,
    default: () => {
      return <h1>I am ArticleContainer</h1>;
    },
  };
});


afterEach(() => {
  cleanup()
});

describe("App", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Pages", () => {
  let fragment;
  let route;

  beforeEach(() => {
    const history = createMemoryHistory();
    history.push(route);
    const { asFragment } = render(
      <Router history={history}>
        <Routes />
      </Router>
    );
    fragment = asFragment()
  });

  describe("Home", () => {
    beforeEach(() => {
      route = '/'
    })

    it("matches snapshot", () => {
      expect(fragment).toMatchSnapshot();
    });

    it('renders the Home component', () => {
      expect(screen.getByRole('heading')).toHaveTextContent('I am Home')
    })
  });

  describe("Articles", () => {
    beforeAll(() => {
      route = "/articles";
    });

    it("matches snapshot", () => {
      expect(fragment).toMatchSnapshot();
    });

    it("renders the Articles component", () => {
      expect(screen.getByRole('heading')).toHaveTextContent('I am ArticlesContainer')
    });
  });

  describe("Article", () => {
    beforeAll(() => {
      route = "/article/1";
    });

    it("matches snapshot", () => {
      expect(fragment).toMatchSnapshot();
    });

    it("renders the Article component", () => {
      expect(screen.getByRole('heading')).toHaveTextContent('I am Article')
    });
  });
});
