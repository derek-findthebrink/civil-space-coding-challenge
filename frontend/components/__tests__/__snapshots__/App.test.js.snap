// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <div
      class="layout__accent-bar"
    />
    <h1>
      I am Home
    </h1>
  </main>
</DocumentFragment>
`;

exports[`Pages Article matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    I am ArticleContainer
  </h1>
</DocumentFragment>
`;

exports[`Pages Articles matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    I am ArticlesContainer
  </h1>
</DocumentFragment>
`;

exports[`Pages Home matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    I am Home
  </h1>
</DocumentFragment>
`;
